src/builtins.c

```diff
@@ -0,0 +1,154 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   builtins.c                                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: dbejar-s <dbejar-s@student.hive.fi>        +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/08/06 14:24:52 by dbejar-s          #+#    #+#             */
+/*   Updated: 2024/08/08 01:46:29 by dbejar-s         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+extern int	g_state;
+
+//si gestionamos g_state, esta funcion deberia retornar un int
+
+void    ft_pwd(void) 
+{
+    char *path;
+
+    path = getcwd(NULL, 0);
+    if (path == NULL)
+    {
+        perror("Error getting current directory");
+        g_state = 1;
+    }
+    else
+    {
+        ft_putendl_fd(path, STDOUT_FILENO);
+        free(path);
+    }
+    return ;
+}
+
+char    *char_pwd(void) 
+{
+    char *path;
+
+    path = getcwd(NULL, 0);
+    if (path == NULL)
+    {
+        perror("Error getting current directory");
+        g_state = 1;
+    }
+    return (path);
+}
+
+void    ft_echo(char *line)
+{
+    int i;
+
+    i = 4;
+    while (line[i] == ' ')
+        i++;
+    while (line[i])
+    {
+        ft_putchar_fd(line[i], STDOUT_FILENO);
+        i++;
+    }
+    ft_putchar_fd('\n', STDOUT_FILENO);
+}
+
+void    ft_echo_n(char *line)
```

Comment:
still don't see the point of two functions xD. I will handle it myself with the help of the tokens and merge

Reply:
If you add to the function is_builtin the following lines, it would be ok, right? It should be before the "echo" check.
	if (ft_strcmp(token->value, "echo -n") == 0)
		return (true);